### PR TITLE
Call checkValidity and provide min-max parameters

### LINKED ISSUE #434: Prepare Omen condition: issues with the Resolution date field
Related to #415, #304

1. It is allowed to enter a year in the distant future - see the pic

2. System allows to enter a day that does not exist in the not leap year

**Expected result:** disable manual input in the field (in Omen, user is allowed to select a date only using a calendar)

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #434

The same problem was in the filters. For prepare condition the min date is set to today.

### CODE REVIEW

File: src/pages/PrepareCondition/index.tsx [hunk 1]
```diff
@@ -541,10 +546,21 @@ export const PrepareCondition = () => {
...
-                      }
+                      onChange={(e) => {
+                        clearErrorOmenCondition('resolutionDate')
+                        if (e.target.checkValidity()) {
+                          setValueOmenCondition('resolutionDate', e.target.value, true)
+                        } else if (e.target.valueAsDate) {
+                          setErrorOmenCondition(
+                            'resolutionDate',
+                            'validity',
+                            'Invalid date or out of range'
```
Comment (hpmaxi) [inline src/pages/PrepareCondition/index.tsx:560]: Is not the same message as in filters ...  I should unify them after review.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr491--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
It works, but I'd suggest that you let the user know which are the min and max valid dates.

For example, for the filters: if the user manually enters and invalid minimum date, use something like "Minimum 'From' date is `so and so`". And do the same for the maximum date.

For "Prepare Condition" it could be simpler: "Valid dates are from today (`date here`) to `maximum date`)."

We shouldn't keep the user guessing.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr491--conditionaltokensexplorer.review.gnosisdev.com

**Comment by elena-zh** [comment]
@hpmaxi , 
I also have reviewed the changes. 
Everything works as expected in calendar filters in the Condition details page. However, some issues are still reproducible on the Prepare condition page: 
- no validation when a date in the past and when a date is incorrect. Actually, validation message appears in a certain cases, and then disappears https://drive.google.com/file/d/1vCQrZpfw6f32UmZ1ZAROqn3SHpc4H1mo/view
-the same with the date in the future: sometimes I am able to see the validation message, sometimes - not
future

Please, take a look into these issues

**Review by mariano-aguero — APPROVED** [review]

**Comment by mariano-aguero** [comment]
@hpmaxi fix the issue that Elena reported. Feel free to merge after that

**Review by gabitoesmiapodo — APPROVED** [review]
Fix is OK. I'd probably go with a shorter message, and we probably should use the same format in the input field and the error message though.

Nevermind, I'll change it later.

**Comment by elena-zh** [comment]
Verified. 
Notes are [here](https://github.com/gnosis/conditional-tokens-explorer/issues/434#issuecomment-713416553)

**Pull request merged**